testing/internal-testing/src/main/groovy/org/gradle/integtests/fixtures/TestClassExecutionResult.java

```diff
@@ -37,6 +38,23 @@ public interface TestClassExecutionResult {
 
     TestClassExecutionResult assertTestCount(int tests, int failures);
 
+    /**
+     * Asserts that the provided metadata appears in the result file at the class level
+     */
+    default TestClassExecutionResult assertMetadata(Map<String, String> props) {
+        assert false;
```

Comment:
These should be using JUnit asserts, unless you mean to fill them in before merging. As-is, they use Java `assert` which is only present with `-ea`, which I don't know if it's set for our tests.